File: apis/elysium-chat-api/src/main/java/com/seventh_root/elysium/api/chat/ChatGroupProvider.java
package com.seventh_root.elysium.api.chat;

import com.seventh_root.elysium.core.service.ServiceProvider;

public interface ChatGroupProvider<T extends ElysiumChatGroup> extends ServiceProvider {

    T getChatGroup(int id);

    T getChatGroup(String name);

    void addChatGroup(T chatGroup);

    void removeChatGroup(T chatGroup);

    void updateChatGroup(T chatGroup);

}

File: apis/elysium-chat-api/src/main/kotlin/com/seventh_root/elysium/api/chat/ChatGroupProvider.kt
package com.seventh_root.elysium.api.chat

import com.seventh_root.elysium.core.service.ServiceProvider

interface ChatGroupProvider<T : ElysiumChatGroup> : ServiceProvider {

    fun getChatGroup(id: Int): T
    fun getChatGroup(name: String): T
    fun addChatGroup(chatGroup: T)
    fun removeChatGroup(chatGroup: T)
    fun updateChatGroup(chatGroup: T)

}
